api/reporting/schemas/reporting-codegen.rb

```diff
@@ -136,68 +261,190 @@ def to_swagger_property(column)
+          else
+            if column[:is_enum]
+              column[:default_enum]
+            else
+              index.to_s
+            end
+          end
+  "public static final #{column[:java_type]} #{column[:java_constant_name]} = #{value};"
+end
+
+constants = COLUMNS.enum_for(:each_with_index) \
+  .map { |col, index| to_constant_declaration(col, index) } \
+  .join("\n")
+
+write_output(OUTPUTS[:unit_test_constants], constants, 'Unit Test Constants')
+
+### Mock Instantiation
+# Mock the projection interface for testing with mock services exposing them
+mocks = COLUMNS.map { |col|
+  "doReturn(#{col[:java_constant_name]}).when(#{TABLE_INFO[:mock]}).#{col[:getter]}();"
+}
+
+lines = []
+lines << "final #{TABLE_INFO[:projection_interface]} #{TABLE_INFO[:mock]} = mock(#{TABLE_INFO[:projection_interface]}.class);"
+lines << mocks
+lines.flatten!
+
+write_output(OUTPUTS[:unit_test_mocks], lines.join("\n"), 'Unit Test Mocks')
+
+### Assertions
+dto_assertions = COLUMNS.map{ |col|
+  getter_call = "#{to_camel_case(table_name, false)}.#{col[:getter]}()"
+  expected = col[:java_constant_name]
+  if col[:java_type].eql?('Timestamp')
+    "    assertTimeApprox(#{getter_call}, #{expected});"
+  else
+    "    assertThat(#{getter_call}).isEqualTo(#{expected});"
+  end
+
+}.join("\n")
+
+write_output(OUTPUTS[:dto_assertions], dto_assertions, 'Unit Test DTO Assertions')
+
+### Parameter Column Enum
+def object_value_function(col)
+  if col[:is_enum]
+    "#{col[:lambda_var]} -> enumToString(#{col[:lambda_var]}.#{col[:getter]}())"
+  else
+    case col[:java_type]
+    when 'Timestamp' # actually OffsetDateTime on the DTO
+      "#{col[:lambda_var]} -> toInsertRowString(#{col[:lambda_var]}.#{col[:getter]}())"
+    else
+      "#{TABLE_INFO[:dto_class]}::#{col[:getter]}"
+    end
+  end
+end
+
+# enum types are special
+JAVA_TYPE_TO_QPV_FACTORY = {
+    'Integer' => 'QueryParameterValue.int64',
+    'Long' => 'QueryParameterValue.int64',
+    'Short' => 'QueryParameterValue.int64',
+    'String' => 'QueryParameterValue.string',
+    'Double' => 'QueryParameterValue.float64',
+    'Boolean' => 'QueryParameterValue.bool',
+    'Timestamp' => 'toTimestampQpv'
+}
+
+def qpv_function(col)
+  if col[:is_enum]
+    convert_fn = 'enumToQpv'
+  else
+    convert_fn = "#{JAVA_TYPE_TO_QPV_FACTORY[col[:java_type]]}"
+  end
+  "#{col[:lambda_var]} -> #{convert_fn}(#{col[:lambda_var]}.#{col[:getter]}())"
+end
+
+def query_parameter_column_entry(col)
+  "#{col[:name].upcase}(\"#{col[:name]}\", #{object_value_function(col)}, #{qpv_function(col)})"
 end
 
-sql = to_query(table_name, columns)
-IO.write(outputs[:projection_query], sql)
-puts "  Projection Query: #{outputs[:projection_query]}"
+QPC_ENUM = (COLUMNS.map do |col|
+  query_parameter_column_entry(col)
+end.join(",\n")).freeze
+
+write_output(OUTPUTS[:query_parameter_COLUMNS], QPC_ENUM, "QueryParameterValue Enum Entries")
+
+### DTO Fluent Setters
```

Comment:
These are just unit test filler lines to instantiate, set values, or do assertions.